tslearn/bases/bases.py

```diff
@@ -237,15 +244,15 @@ def from_json(cls, path):
         Model instance
         """
 
-        model = json.load(open(path, 'r'))
+        model = json.load(open(path, "r"))
         model = cls._byte2string(model)
 
         # Convert the lists back to arrays
-        for param_type in ['model_params', 'hyper_params']:
+        for param_type in ["model_params", "hyper_params"]:
             for k in model[param_type].keys():
                 param = model[param_type][k]
                 if type(param) is list:
-                    arr = np.array(param)
+                    arr = np.array(param, dtype=object)
```

Comment:
Same question here

Reply:
When I remove `dtype=object`, we have the following continuous integration test error:

```
=================================== FAILURES ===================================
___________________________ test_serialize_shapelets ___________________________

    def test_serialize_shapelets():
        def get_model_weights(model):
            return model.model_.get_weights()
    
        n, sz, d = 15, 10, 3
        rng = numpy.random.RandomState(0)
        X = rng.randn(n, sz, d)
    
        for y in [
            rng.randint(low=0, high=3, size=n),
            rng.choice(["one", "two", "three"], size=n),
        ]:
            shp = LearningShapelets(max_iter=1)
            _check_not_fitted(shp)
            shp.fit(X, y)
>           _check_params_predict(
                shp,
                X,
                ["predict"],
                check_params_fun=get_model_weights,
                formats=["json", "pickle"],
            )

tslearn/tests/test_serialize_models.py:289: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tslearn/tests/test_serialize_models.py:90: in _check_params_predict
    sm = getattr(model, "from_{}".format(fmt))(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class 'tslearn.shapelets.shapelets.LearningShapelets'>
path = '/home/vsts/work/1/s/tslearn/tests/tmp/LearningShapelets.json'

    @classmethod
    def from_json(cls, path):
        """
        Load model from a JSON file.
    
        Parameters
        ----------
        path : str
            Full path to file.
    
        Returns
        -------
        Model instance
        """
    
        model = json.load(open(path, "r"))
        model = cls._byte2string(model)
    
        # Convert the lists back to arrays
        for param_type in ["model_params", "hyper_params"]:
            for k in model[param_type].keys():
                param = model[param_type][k]
                if type(param) is list:
>                   arr = np.array(param)
E                   ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape was (20,) + inhomogeneous part.

tslearn/bases/bases.py:255: ValueError
```